src/Main.elm

```diff
@@ -155,12 +228,24 @@ update msg model =
                     ( { model
                         | focusedMovie = Movie.Loaded movie
                       }
-                    , Cmd.none
+                    , searchJustWatch movie
                     )
 
                 _ ->
                     ( model, Cmd.none )
 
+        LoadJustWatchDetails movieDetails ->
+            -- when results are loaded, we don't want to use them unless the movie is still focused
+            case model.focusedMovie of
+                Movie.Loaded focusedMovie ->
+                    if movieDetails.movie.title == focusedMovie.movie.title then
```

Comment:
Is this because the just watch search  might return a partial match that isn't the exact movie?

Reply:
This is for the case I mentioned in chat where the response might come back after the focused movie has changed - user closed dialog or picked another movie.

Reply:
Oh, right. I forgot about the case where the user opened a new movie.

I think this approach is good. No objections.  That said, since we're (probably) not worried about movie details becoming stale, tthere's another approach that might also work.  We could add the movie details to `Movie` as `details: Maybe MovieDetails` (defaults to `Nothing`). Then when this message comes back, instead of updating `focusedMovie`, we just update the movie in the full list, and pull that out in the view.  Or something generally like that - I might have mixed up types a bit because I'm writing this while watching a roadmap presentation.

Reply:
Thanks for the idea. Something to work on at the next meetup!